src/api-v1.js

```diff
@@ -63,35 +64,35 @@ api.declare({
   let logthingy = `${url} in ${service}/${region}`;
   debug(`Attempting to redirect to ${logthingy}`);
 
+  try {
+    await validateInputUrl(url, this.allowedPatterns, this.redirectLimit, this.ensureSSL);
+  } catch (err) {
+    debug(err.stack || err);
+    // Intentionally vague because we don't want to reveal
+    // our configuration too much
+    let msg = 'Input URL failed validation for unknown reason';
```

Comment:
This is a small thing, but generally avoid using exceptions to signal result.

In this case
```js
let msg = await validateInputUrl(...);
if (msg) {
  return res.status(400).json({msg});
}
```

Typically, a function has two outputs the _result_ or the _exception_. In `validateInputUrl` it's easy to think that a validation error is an exception. But really that is the main thing this function does, so it's not an exception, it's the _result_.

A good way to see this is that an _invalid url_ is different from a url we were _unable to validate_.
If the url is invalid, we return 400, if an unknown error happens during validation we return 500. This also remove the catch branch.